UpdatedSyntheticDataset/SyntheticDataset2/ImageCreator/target_creator.py

```diff
@@ -0,0 +1,211 @@
+from PIL import Image
+import math
+from SyntheticDataset2.ElementsCreator import *
+from SyntheticDataset2.ImageOperations import *
+
+class TargetCreator(object):
+    """
+    :param shapelist: the list of shapes that are going to be used. Orientations are separated.
+        indexes:
+        0 = circle
+
+        1 = quarter circle with corner facing up
+        2 = quarter circle with corner facing up right
+        3 = quarter circle with corner facing right
+        4 = quarter circle with corner facing down right
+        5 = quarter circle with corner facing down
+        6 = quarter circle with corner facing down left
+        7 = quarter circle with corner facing left
+        8 = quarter circle with corner facing up left
+
+        9 = half circle with side facing up
+        10 = half circle with side facing right
+        11 = half circle with side facing down
+        12 = half circle with side facing left
+
+        13 = cross with convex facing up
+        14 = cross with concave facing up
+
+        15 = triangle with angle facing up
+        16 = triangle with side facing up
+
+        17 = square with angle facing up
+        18 = square with side facing up
+
+        19 = rectangle with shorter side facing up
+        20 = rectangle with longer side facing up
+
+        21 = trapezoid with shorter side facing up
+        22 = trapezoid with longer side facing up
+
+        23 = pentagon with angle facing up
+        24 = pentagon with side facing up
+
+        25 = hexagon with angle facing up
+        26 = hexagon with side facing up
+
+        27 = heptagon with angle facing up
+        28 = heptagon with side facing up
+
+        29 = octagon with angle facing up
+        30 = octagon with side facing up
+
+        31 = star with convex facing up
+        32 = star with concave facing up
+    """
```

Comment:
Are you sure that you have to write a whole shape index?

Is it possible that you could have the methods take a `shape_name` and `orientation` var?

Reply:
I can make a shape_name parameter instead of the shape_index parameter. But the effect would be the same, you still need to remember the shapes' names. If you want a semicircle with angle facing top right, I guess you need to put "semicircle" as shape_name and "angle facing top right" as orientation? Then if you want all of these to be random, you also need a list of all shapes and orientations to do that.

Reply:
James is correct this is not clean code. You should make a static class called ShapeTypes and a static class called OrientationTypes to hold each of these values. You then set constants with good names (such as STAR, SQUARE, NW, W, etc) which each correspond to a number. You then compare the input to each of those constants to see if they are the same.

This way, to the viewer everything makes sense.

Reply:
What @DarkmatterVale said—I agree.